src/Public/Security.ps1

```diff
@@ -122,14 +128,40 @@ function Add-PodeSecurityHeader {
 
         [Parameter()]
         [string]
-        $Value
+        $Value,
+
+        [Parameter()]
+        [switch]
+        $Add,
+
+        [Parameter()]
+        [ValidateSet('Upper', 'Lower')]
+        [string]
+        $Casing
     )
 
     if (![string]::IsNullOrWhiteSpace($Value)) {
-        $PodeContext.Server.Security.Headers[$Name] = $Value
+        if ($add -and $PodeContext.Server.Security.Headers.ContainsKey($Name)) {
+            $Headers = $PodeContext.Server.Security.Headers[$Name].split(',')
+            if ($Headers -contains $Value) {
+                return
+            } else {
+                $Headers += $Value
+                $PodeContext.Server.Security.Headers[$Name] = (($Headers.trim() | Select-Object -Unique) -join ', ')
+            }
+        } else {
+            $PodeContext.Server.Security.Headers[$Name] = $Value
+        }
+
+        if ($Casing) {
+            if ($Casing -ieq 'Upper') {
+                $PodeContext.Server.Security.Headers[$Name] = $PodeContext.Server.Security.Headers[$Name].ToUpperInvariant()
+            } elseif ($Casing -ieq 'Lower') {
+                $PodeContext.Server.Security.Headers[$Name] = $PodeContext.Server.Security.Headers[$Name].ToLowerInvariant()
+            }
+        }
```

Comment:
I was thinking about this yesterday, I think the casing can be dropped completely - the Headers, and other values, should really be in the same casing that somebody entered. It should all be case insensitive anyway, so saves 7 lines of code 😂

Reply:
but it's looking better when you debug CORS issues :)